pkg/route/validation/validation.go

```diff
@@ -342,6 +385,53 @@ func validateWildcardPolicy(host string, policy routev1.WildcardPolicyType, fldP
 	return nil
 }
 
+var (
+	notAllowedHTTPHeaders   = []string{"strict-transport-security", "proxy", "cookie", "set-cookie"}
+	notAllowedHTTPHeaderSet = sets.NewString(notAllowedHTTPHeaders...)
+)
+
+func validateHeaders(path *field.Path, headerValues []routev1.RouteHTTPHeader) *field.Error {
+	unique := make(map[string]bool, len(headerValues))
+	us := make([]string, len(unique))
+	for key, value := range headerValues {
+		if len(value.Name) == 0 {
+			return field.Invalid(path.Index(key).Child("name"), value.Name, fmt.Sprintf("name must be present"))
+		}
+
+		if !unique[value.Name] {
+			us = append(us, value.Name)
+			unique[value.Name] = true
+		} else {
+			return field.Invalid(path.Index(key).Child("name"), value.Name, fmt.Sprintf("duplicate header name not allowed"))
+		}
+
+		if value.Action.Set != nil && len(value.Action.Set.Value) > 16384 {
```

Comment:
nit: again, this numeric value should really be a constant var

Reply:
ok